pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -112,12 +116,34 @@ func NewController(client kube.Client, meshWatcher mesh.Holder,
 		controllers.WithReconciler(c.onEvent),
 		controllers.WithMaxAttempts(5))
 	c.ingress.AddEventHandler(controllers.ObjectHandler(c.queue.AddObject))
+
+	// We watch service changes to detect service port number change to trigger
+	// re-convert ingress to new-vs.
+	c.serviceQueue = controllers.NewQueue("service",
```

Comment:
each controller should only have one queue. Rather than doing `c.serviceQueue.Add(o)`, it should do some operation to get the Ingress objects impacted. Then it should do `c.queue.Add(ingress)`.

it should not call `notifyVirtualServiceHandler`

Reply:
Do you mean that we should use single queue to handle ingress and service resources? In the processing of service event, find all relative ingress resources and enqueue these ingress resources to the following process. Is my understanding correct？